### DISCUSSION TITLE
Perfect Physics Simulation

### DISCUSSION THREAD

**Discussion by CarlKCarlK** [pyscript/pyscript#1423] [Show and tell]
Thanks to @JeffersGlass for helping to get this simulator working on PyScript. The project uses the SymPy computer algebra package to simulate Newtonian physics "perfectly". 

It is perfect in the sense that it represents all values (for example, velocity, time) as expressions rather than numeric approximations. 

https://carlkcarlk.github.io/perfect-physics/pyscript/

- Carl
p.s. I'll be presenting the project tomorrow morning at the PyData conference in Seattle. If you are there, too, say "hi!".

**Comment by hoodmane** [comment]
You may want to suppress these warnings:
```
/lib/python3.10/site-packages/perfect_physics/_misc.py:97: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).
  new_manager = plt.figure().canvas.manager
WARNING:root:Multiple collisions: 2
WARNING:root:Multiple collisions: 2
```
